File: src/main/java/com/ggardet/patient/spi/dto/PatientDTO.java
package com.ggardet.patient.spi.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;
import java.util.UUID;

public record PatientDTO(
        UUID id,
        @NotBlank String firstname,
        @NotBlank String lastname,
        @NotNull LocalDate birthDate,
        String socialSecurityNumber,
        String phoneNumber,
        @Email String email,
        UUID hospitalId,
        UUID serviceId
) {}

File: src/main/kotlin/com/ggardet/patient/spi/dto/PatientDTO.kt
package com.ggardet.patient.spi.dto

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import java.time.LocalDate
import java.util.*

data class PatientDTO(
    val id: UUID? = null,
    @field:NotBlank val firstname: String,
    @field:NotBlank val lastname: String,
    @field:NotNull val birthDate: LocalDate,
    val socialSecurityNumber: String? = null,
    val phoneNumber: String? = null,
    @field:Email val email: String? = null,
    val hospitalId: UUID? = null,
    val serviceId: UUID? = null
)